### PR TITLE
update geo related deps.

### PR DESCRIPTION
**Pull request by michaelkirk**
geo-types to 0.7
geo to 0.17 which removes redundant old geo used by polylabel
geojson to 0.22

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -36,9 +36,18 @@ members = [
...
 [patch.crates-io]
-# Some niceties for parsing geojson feature properties.
-# Upstreaming at https://github.com/georust/geojson/pull/155
-geojson = { git = "https://github.com/georust/geojson" }
 # Since we've updated our tokio runtime to 1.0 elsewhere, we need to use a compatible version of flatgeobuf, until this is released:
 # https://github.com/flatgeobuf/flatgeobuf/commit/b21bfc72396505ce564b3195d59e63d22a48454f#diff-73865c8d3c63ba309958fca980e3f4d75ec4aa20df8d58b411479e44880f39ec
 flatgeobuf = { git = "https://github.com/flatgeobuf/flatgeobuf" }
+
+# update to geo-types 0.7 - upstream: https://github.com/21re/rust-geo-booleanop/pull/27
+geo-booleanop = { git = "https://github.com/michaelkirk/rust-geo-booleanop", branch = "mkirk/geo-types-0.7" }
```
Comment (michaelkirk) [inline Cargo.toml:44]: I had to open several upstream PR's. If you'd rather wait until these are merged, that's also fine with me. 

This exercise with abstreet was a good impetus to get them opened anyway...

File: Cargo.toml [hunk 2]
```diff
@@ -36,9 +36,18 @@ members = [
...
+
+# update to geo-types 0.7 - upstream: https://github.com/georust/geozero/pull/5
+geozero-core = { path = "../../georust/geozero/geozero-core", branch = "mkirk/geo-types-0.7" }
+
+# update to geo-types 0.7 - upstream: https://github.com/mthh/contour-rs/pull/2
+contour = { git = "https://github.com/michaelkirk/contour-rs", branch = "mkirk/geojson-0.22" }
+
+# update to geo-types 0.7 not released yet
+gdal = { git = "https://github.com/georust/gdal" }
```
Comment (michaelkirk) [inline Cargo.toml:53]: I believe gdal is only used when running `./import.sh --scenario`

I tried to test the import script. It compiled, but when running, I was met with:

```
- Missing data/input/seattle/N47W122.hgt, so downloading https://dds.cr.usgs.gov/srtm/version2_1/SRTM1/Region_01/N47W122.hgt.zip
- Running "curl" "--fail" "-L" "-o" "tmp_output" "https://dds.cr.usgs.gov/srtm/version2_1/SRTM1/Region_01/N47W122.hgt.zip"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404 Not Found
thread 'main' panicked at '"curl" "--fail" "-L" "-o" "tmp_output" "https://dds.cr.usgs.gov/srtm/version2_1/SRTM1/Region_01/N47W122.hgt.zip" failed', abstutil/src/process.rs:9:17
```

Which looks unrelated, just FYI, seems like some file we expect to be there has been moved/deleted.

File: Cargo.toml (hunk 1 above)
Comment (dabreegster) [reply Cargo.toml:44]: I'm happy to do these updates now and immediately reap the smaller dependencies! Thanks for going through the chore of fixing everything upstream.

File: Cargo.toml (hunk 2 above)
Comment (dabreegster) [reply Cargo.toml:53]: Whoops, good catch. This file is already downloaded on my end, so I haven't noticed it vanish. Not a big concern since most people aren't generating Seattle data from scratch, but I'll do something about this. Maybe just temporarily disabling this, since the elevation parsing code is incorrect anyway and unused.

**Comment by dabreegster** [comment]
Nice to see just one version of geo. Thank you!

File: Cargo.toml [hunk 3]
```diff
@@ -36,9 +36,18 @@ members = [
...
-geojson = { git = "https://github.com/georust/geojson" }
 # Since we've updated our tokio runtime to 1.0 elsewhere, we need to use a compatible version of flatgeobuf, until this is released:
 # https://github.com/flatgeobuf/flatgeobuf/commit/b21bfc72396505ce564b3195d59e63d22a48454f#diff-73865c8d3c63ba309958fca980e3f4d75ec4aa20df8d58b411479e44880f39ec
 flatgeobuf = { git = "https://github.com/flatgeobuf/flatgeobuf" }
+
+# update to geo-types 0.7 - upstream: https://github.com/21re/rust-geo-booleanop/pull/27
+geo-booleanop = { git = "https://github.com/michaelkirk/rust-geo-booleanop", branch = "mkirk/geo-types-0.7" }
+
+# update to geo-types 0.7 - upstream: https://github.com/georust/geozero/pull/5
+geozero-core = { path = "../../georust/geozero/geozero-core", branch = "mkirk/geo-types-0.7" }
```
Comment (dabreegster) [inline Cargo.toml:47]: Whoops, I should test PRs locally first. :P  This path should be https://github.com/michaelkirk/geozero/tree/mkirk/geo-types-0.7, I'll fix

Comment (dabreegster) [reply Cargo.toml:47]: https://github.com/a-b-street/abstreet/commit/783ce53a4eaede83bb19c880ce1f2e1b0af14012 Easy fix

Comment (michaelkirk) [reply Cargo.toml:47]: blech! sorry.
